src/inmanta/ast/statements/assign.py

```diff
@@ -555,16 +555,36 @@ def __repr__(self) -> str:
         )
 
 
-class StringFormat(ReferenceStatement):
+class FormattedString(ReferenceStatement):
     """
-    Create a new string by doing a string interpolation
+    This class is an abstraction around a string containing references to variables.
     """
 
     __slots__ = ("_format_string", "_variables")
 
-    def __init__(self, format_string: str, variables: typing.List[typing.Tuple["Reference", str]]) -> None:
-        ReferenceStatement.__init__(self, [k for (k, _) in variables])
+    def __init__(self, format_string: str, variables: Sequence["Reference"]) -> None:
+        super().__init__(variables)
         self._format_string = format_string
+
+    def execute(self, requires: typing.Dict[object, object], resolver: Resolver, queue: QueueScheduler) -> object:
+        return super().execute(requires, resolver, queue)
+
+    def get_dataflow_node(self, graph: DataflowGraph) -> dataflow.NodeReference:
+        return dataflow.NodeStub("StringFormat.get_node() placeholder for %s" % self).reference()
+
+    def __repr__(self) -> str:
+        return "Format(%s)" % self._format_string
+
+
+class StringInterpolationFormat(FormattedString):
```

Comment:
```suggestion
class StringFormat(FormattedString):
```

I would not change the name if there is no strong reason. This might be used in code somewhere.